### ISSUE TITLE
Support for disabling certain of the embed link type

### ISSUE THREAD

**Issue by Antimated** [verbb/hyper#24]
### What are you trying to do?

Hi 👋 

I'm trying some things out with hyper and one idea I had was creating a "Video embed" matrix block which well... displays embedded video's. The only thing I could't do was set a list of certain providers I want to be entered by the customer. In this case I would only want to whitelist "Youtube" and "Vimeo".

### What's your proposed solution?

Whitelist (or blacklist) for enabling/disabling certain types of embeds.

**Comment by engram-design** [comment]
Yep, so at the moment, an embed link type just accepts _anything_, we can look at maybe adding settings for allow/deny providers, but we also don't have a definitive list of providers to select from. We also can't use the base domain because providers like YouTube use multiple ones.

**Comment by benfeather** [comment]
I would like to see this feature implemented as well. Is it possible to have a simple regex input in the field settings? This way, the user can configure it to match any site, and there would be no need for you to maintain an exhaustive list of providers.

**Comment by engram-design** [comment]
Added for the next release. To get this early, run `composer require verbb/hyper:"dev-craft-4 as 1.1.17"`.

**Comment by benfeather** [comment]
It's probably really obvious, but I can't seem to find where to update the field.

**Comment by engram-design** [comment]
It's not a field setting, but a [plugin setting](https://verbb.io/craft-plugins/hyper/docs/get-started/configuration). See `embedAllowedDomains`

**Comment by benfeather** [comment]
Could this setting be editable at the field level? 

This would allow the user to create a field for each embed type, e.g. a video field would whitelist YouTube/Vimeo etc. Then, it would be easy to know what data is stored in the new linkValue JSON property, as it will be consistent for each provider. 

Just an idea, though not essential.

**Comment by engram-design** [comment]
Added in [3.0.0-beta.1](https://github.com/verbb/hyper/releases/tag/3.0.0-beta.1).
